packages/components/src/mobile/keyboard-avoiding-view/index.ios.js

```diff
@@ -3,19 +3,103 @@
  */
 import {
 	KeyboardAvoidingView as IOSKeyboardAvoidingView,
+	Animated,
+	Keyboard,
 	Dimensions,
+	View,
 } from 'react-native';
 
-export const KeyboardAvoidingView = ( { parentHeight, ...otherProps } ) => {
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { useResizeObserver } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+
+const AnimatedKeyboardAvoidingView = Animated.createAnimatedComponent(
+	IOSKeyboardAvoidingView
+);
+
+const MIN_HEIGHT = 44;
+
+export const KeyboardAvoidingView = ( {
+	parentHeight,
+	style,
+	withAnimatedHeight = false,
+	...otherProps
+} ) => {
+	const [ keyboardHeight, setKeyboardHeight ] = useState( 0 );
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const { height = 0 } = sizes || {};
+
+	const animatedHeight = useRef( new Animated.Value( MIN_HEIGHT ) ).current;
+
 	const { height: fullHeight } = Dimensions.get( 'window' );
 	const keyboardVerticalOffset = fullHeight - parentHeight;
 
+	useEffect( () => {
+		Keyboard.addListener( 'keyboardWillShow', onKeyboardWillShow );
+		Keyboard.addListener( 'keyboardWillHide', onKeyboardWillHide );
+		return () => {
+			Keyboard.removeListener( 'keyboardWillShow', onKeyboardWillShow );
+			Keyboard.removeListener( 'keyboardWillHide', onKeyboardWillHide );
+		};
+	}, [] );
+
+	useEffect( () => {
+		animate();
+	}, [ keyboardHeight ] );
+
+	function onKeyboardWillShow( { endCoordinates } ) {
+		setKeyboardHeight( endCoordinates.height );
+	}
+
+	function onKeyboardWillHide() {
+		setKeyboardHeight( 0 );
+	}
+
+	const paddedKeyboardHeight =
+		keyboardHeight + MIN_HEIGHT - ( style.bottom || 0 );
+
+	function animate() {
+		if ( keyboardHeight ) {
+			animatedHeight.setValue( paddedKeyboardHeight );
+		} else
+			Animated.timing( animatedHeight, {
+				toValue: MIN_HEIGHT,
+				duration: 150,
```

Comment:
Just wondering why it is `150`? :)

Reply:
When I tested it before, this value seemed to be correct to me, however, I have decided to set the value to `200` and compare it with the hiding time of the keyboard and have to say that `200` is the better choice.

<img width="300" src="https://user-images.githubusercontent.com/22746080/89560461-4ca5c080-d817-11ea-8a84-9cd49e9e4583.gif" />

Reply:
I believe you could take a `duration` from Keyboard event :)

Reply:
Done - passed duration from Keyboard event!